CharlieBackend.Data/Repositories/Impl/AccountRepository.cs

```diff
@@ -60,13 +60,34 @@ public void UpdateAccountCredentials(Account account)
             _applicationContext.Entry(account).Property(a => a.Salt).IsModified = true;
         }
 
-        public async Task<bool> IsEmailChangableToAsync(string newEmail)
+        public async Task<bool> IsEmailChangableToAsync(long? id, string newEmail)
         {
             var count = await _applicationContext.Accounts
                     .Where(account => account.Email == newEmail)
                     .CountAsync();
-            if (count > 1) return false;
-            return true;
+
+            if (count == 1)
+            {
+                var foundAccountOfEmail = await _applicationContext.Accounts
+                    .FirstOrDefaultAsync(account => account.Email == newEmail);
+
+                if (foundAccountOfEmail.Id == id)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            else if (count > 1)
```

Comment:
How can we have more than one account with particular email?

Reply:
In previous implementation of this method develovers allowed >1 email in accounts. So i have saved this from previous development.
I agree, i will remove.